### PR TITLE
Port Pack Docs changes to 3.7

### PR DESCRIPTION
**Pull request by aeisenberg**
This moves the following three PRs to the 3.7 branch:

- https://github.com/github/codeql/pull/10182
- https://github.com/github/codeql/pull/10146
- https://github.com/github/codeql/pull/10105

Also, this adds a changelog entry for some new features that were undocumented.

### CODE REVIEW

File: docs/codeql/codeql-cli/creating-codeql-databases.rst [hunk 1]
```diff
@@ -210,11 +210,10 @@ commands that you can specify for compiled languages.
...
 - C# project built using ``dotnet build``::
 
-     For C# projects using either `dotnet build` or `msbuild`, you should specify `/p:UseSharedCompilation=false`
-     in the build command. It is also a good idea to add `/t:rebuild` to ensure that all code will be built (code
-     that is not built will not be included in the CodeQL database):
+     It is a good idea to add `/t:rebuild` to ensure that all code will be built, or do a
+     prior `dotnet clean` (code that is not built will not be included in the CodeQL database):
 
-     codeql database create csharp-database --language=csharp --command='dotnet build /p:UseSharedCompilation=false /t:rebuild' 
+     codeql database create csharp-database --language=csharp --command='dotnet build /t:rebuild'
```
Comment (aeisenberg) [inline docs/codeql/codeql-cli/creating-codeql-databases.rst:217]: This change is not directly related to the packaging change. I am not sure if it should be backported.

File: docs/codeql/codeql-cli/creating-codeql-databases.rst [hunk 2]
```diff
@@ -349,8 +348,7 @@ The following example shows how you could use indirect build tracing in an Azure
        - task: VSBuild@1
            inputs:
              solution: '**/*.sln'
-             # Disable MSBuild shared compilation for C# builds.
-             msbuildArgs: /p:OutDir=$(Build.ArtifactStagingDirectory) /p:UseSharedCompilation=false
```
Comment (aeisenberg) [inline docs/codeql/codeql-cli/creating-codeql-databases.rst:353]: Same

**Comment by aeisenberg** [comment]
I don't understand this error: https://github.com/github/semmle-code/actions/runs/3023999175/jobs/4866273380#step:17:78

File: docs/codeql/codeql-cli/creating-codeql-databases.rst (hunk 1 above)
Comment (henrymercer) [reply docs/codeql/codeql-cli/creating-codeql-databases.rst:217]: @criemen can you advise?

Comment (criemen) [reply docs/codeql/codeql-cli/creating-codeql-databases.rst:217]: @hvitved as far as I'm aware, the code changes for this docs change is in `rc/3.7`, and therefore, we should also backport the docs changes. Can you please confirm that this is correct?

**Comment by aeisenberg** [comment]
Still failing. Not sure how to fix. Looks like the SHAs found for the `useful-deleted-files.sh` script are incorrect. I am not sure why this script is failing, but others are passing.

Relevant job: https://github.com/github/semmle-code/actions/runs/3038886452/jobs/4893158954

Triggered by: `/qlucie ql --test-pr 10367 --code-ref aeisenberg/fix-ghprbtargetbranch`

**Comment by aeisenberg** [comment]
Looking a little deeper, it seems like a bunch of the checks failed, but they did not cause the job to fail.

File: docs/codeql/codeql-cli/creating-codeql-databases.rst (hunk 1 above)
Comment (hvitved) [reply docs/codeql/codeql-cli/creating-codeql-databases.rst:217]: Yes, should be ok to backport.

**Review by henrymercer — APPROVED** [review]
LGTM.  Looks like the checks triggered directly by the PR are passing now we've merged the internal fix.
